### PR TITLE
Ensure max length

### PR DESCRIPTION
**Pull request by shargon**
Related to: https://github.com/neo-project/neo-modules/pull/827#issuecomment-1815273757

### CODE REVIEW

**Review by Liaojinghui — APPROVED** [review]

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -166,6 +166,11 @@ private static JObject ToJson(StackItem item, Session session)
 
         private static Signer[] SignersFromJson(JArray _params, ProtocolSettings settings)
         {
+            if (_params.Count > Transaction.MaxTransactionAttributes)
+            {
+                throw new RpcException(-100, "Max allowed witness exceeded.");
```
Comment (AnnaShaleva) [inline src/RpcServer/RpcServer.SmartContract.cs:171]: > We can wait for https://github.com/neo-project/neo-modules/pull/815

Agree, some InvalidParams error can be used here. I'll review #815 one more time ASAP.

**Comment by cschuchardt88** [comment]
have i max allowed witness conditions been fixed in neo-core or here? like `and` `boolean` `not` etc

**Comment by shargon** [comment]
> have max allowed witness conditions been fixed in neo-core or here? like `and` `boolean` `not` etc

Must be on both, here is composed manually, not deserialized.

**Comment by Liaojinghui** [comment]
@shargon , grand master, can we have a sole project where all limits are set together? the existing limitations are like here and there, hard to follow and track.

**Comment by shargon** [comment]
> @shargon , grand master, can we have a sole project where all limits are set together? the existing limitations are like here and there, hard to follow and track.

Agree, but the limits are during deserialization, and here the object is composed manually

**Comment by shargon** [comment]
I will merge it, #815 is delayed and it can replace this throw later
